### PR TITLE
Display survey only when dealing with Jupyter Notebooks

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #5181

### CODE REVIEW

File: src/client/datascience/dataScienceSurveyBanner.ts [hunk 1]
```diff
@@ -169,7 +170,15 @@ export class DataScienceSurveyBanner implements IJupyterExtensionBanner, IExtens
     }
 
     public async activate() {
-        this.vscodeNotebook.onDidOpenNotebookDocument(this.openedNotebook, this, this.disposables);
+        this.vscodeNotebook.onDidOpenNotebookDocument(
+            (e) => {
+                if (isJupyterNotebook(e)) {
```
Comment (DonJayamanne) [inline src/client/datascience/dataScienceSurveyBanner.ts:175]: This is the bug fix.

File: src/client/datascience/kernel-launcher/kernelDaemonPreWarmer.ts [hunk 2]
```diff
@@ -88,7 +88,7 @@ export class KernelDaemonPreWarmer {
     // Handle opening of native documents
     private async onDidOpenNotebookDocument(doc: NotebookDocument): Promise<void> {
         // It could be anything, lets not make any assumptions.
-        if (isUntitledFile(doc.uri)) {
+        if (isUntitledFile(doc.uri) || !isJupyterNotebook(doc)) {
```
Comment (DonJayamanne) [inline src/client/datascience/kernel-launcher/kernelDaemonPreWarmer.ts:91]: We don't need to pre-warm kernels when user opens non-jupyter notebooks

File: src/client/datascience/notebook/defaultCellLanguageService.ts [hunk 3]
```diff
@@ -54,6 +54,9 @@ export class NotebookCellLanguageService implements IExtensionSingleActivationSe
     }
     @swallowExceptions('Saving last saved cell language')
     private async onDidSaveNotebookDocument(doc: NotebookDocument) {
+        if (!isJupyterNotebook(doc)) {
+            return;
```
Comment (DonJayamanne) [inline src/client/datascience/notebook/defaultCellLanguageService.ts:58]: We don't care if users save non-Jupyter notebooks

File: src/client/datascience/notebook/notebookEditorProvider.ts [hunk 4]
```diff
@@ -200,6 +200,10 @@ export class NotebookEditorProvider implements INotebookEditorProvider {
             this._onDidChangeActiveNotebookEditor.fire(undefined);
             return;
         }
+        if (isJupyterNotebook(editor.document)) {
+            this._onDidChangeActiveNotebookEditor.fire(undefined);
```
Comment (DonJayamanne) [inline src/client/datascience/notebook/notebookEditorProvider.ts:204]: If user opens a non-jupyter notebook, then our active editor is `undefined`.

File: src/client/datascience/notebook/remoteSwitcher.ts [hunk 5]
```diff
@@ -46,7 +47,7 @@ export class RemoteSwitcher implements IExtensionSingleActivationService {
         await this.serverSelector.selectJupyterURI(true, 'nativeNotebookToolbar');
     }
     private async updateStatusBar() {
-        if (!this.notebook.activeNotebookEditor) {
+        if (!this.notebook.activeNotebookEditor || !isJupyterNotebook(this.notebook.activeNotebookEditor.document)) {
```
Comment (DonJayamanne) [inline src/client/datascience/notebook/remoteSwitcher.ts:50]: We don't care about non-jupyter notebooks.

**Review by IanMatthewHuff — APPROVED** [review]

**Pull request merged**